### PR TITLE
Add method for patch `corpora.Dictionary` based on special tokens

### PR DESCRIPTION
**Pull request by Froskekongen**
First part of fixing this issue: https://github.com/RaRe-Technologies/gensim/issues/2190

This function patches `token2id` in dictionary based on a dict with special tokens and their wanted indices.

### CODE REVIEW

File: gensim/test/test_corpora_dictionary.py [hunk 1]
```diff
@@ -309,6 +309,16 @@ def test_dict_interface(self):
...
 
+    def test_patch_with_special_tokens(self):
+        special_tokens = {'pad': 0, 'space': 1}
+        corpus = [["máma", "mele", "maso"], ["ema", "má", "máma"]]
+        d = Dictionary(corpus)
+        self.assertEqual(len(d.token2id), 5)
+        d.patch_with_special_tokens(special_tokens)
+        self.assertEqual(d.token2id['pad'], 0)
+        self.assertEqual(d.token2id['space'], 1)
+        self.assertEqual(len(d.token2id), 7)
```
Comment (menshikh-iv) [inline gensim/test/test_corpora_dictionary.py]: Need to extend this test (check also than you can transform document with special tokens)

**Comment by Froskekongen** [comment]
@menshikh-iv: I fixed the logic and updated the tests to also account for this case.

File: gensim/test/test_corpora_dictionary.py (hunk 1 above)
Comment (Froskekongen) [reply gensim/test/test_corpora_dictionary.py]: Extension added.

**Comment by menshikh-iv** [comment]
@piskvorky wdyt about this PR?

File: gensim/test/test_corpora_dictionary.py (hunk 1 above)
Comment (Froskekongen) [reply gensim/test/test_corpora_dictionary.py]: It wasn't obvious to me that this was the indention - to test the doc2bow transformation. I have added tests for that now, however.

Comment (menshikh-iv) [reply gensim/test/test_corpora_dictionary.py]: That's strange because this is the main method for an apply dictionary to document to get BoW, thanks!

**Comment by Froskekongen** [comment]
@menshikh-iv: Are you planning to accept this PR?

**Comment by piskvorky** [comment]
I'd prefer to have the docstring expanded. What is the motivation for this function? Who would want to call it and why?

If it's fully backward compatible, we can merge it. It just needs a better description, so it doesn't die forgotten in solitude.

**Comment by menshikh-iv** [comment]
@Froskekongen please do https://github.com/RaRe-Technologies/gensim/pull/2200#issuecomment-429407341 and I'll merge current PR.

**Comment by Froskekongen** [comment]
@menshikh-iv: Updated documentation of the function.

File: gensim/corpora/dictionary.py [hunk 2]
```diff
@@ -612,16 +611,21 @@ def patch_with_special_tokens(self, special_token_dict):
...
-        >>>
-        >>> corpus = [["máma", "mele", "maso"], ["ema", "má", "máma"]]
-        >>> dct = Dictionary(corpus)
-        >>> special_tokens = {'pad': 0, 'space': 1}
-        >>> print(dct.token2id)
-        {'maso': 0, 'mele': 1, 'máma': 2, 'ema': 3, 'má': 4}
-        >>> dct.patch_with_special_tokens(special_tokens)
-        >>> print(dct.token2id)
-        {'maso': 6, 'mele': 7, 'máma': 2, 'ema': 3, 'má': 4, 'pad': 0, 'space': 1}
+        .. sourcecode:: pycon
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/corpora/dictionary.py]: that's special section name for using `flake8` on docstrings

[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/corpora/dictionary.py]: see https://github.com/RaRe-Technologies/gensim/pull/2192

[the remark this answers is not in the document]
Comment (piskvorky) [reply gensim/corpora/dictionary.py]: Aha, OK. Can you update our Developer wiki page with such tricks?

Also (unrelated) with the workflow process you currently use to label and manage issues and PRs (`interesting PR` label etc).

**Comment by menshikh-iv** [comment]
thanks for PR @Froskekongen, congratz with first contribution :1st_place_medal:

File: gensim/corpora/dictionary.py [hunk 3]
```diff
@@ -594,6 +594,52 @@ def merge_with(self, other):
...
+            >>> corpus = [["máma", "mele", "maso"], ["ema", "má", "máma"]]
+            >>> dct = Dictionary(corpus)
+            >>>
+            >>> special_tokens = {'pad': 0, 'space': 1}
+            >>> print(dct.token2id)
+            {'maso': 0, 'mele': 1, 'máma': 2, 'ema': 3, 'má': 4}
+            >>>
+            >>> dct.patch_with_special_tokens(special_tokens)
+            >>> print(dct.token2id)
+            {'maso': 6, 'mele': 7, 'máma': 2, 'ema': 3, 'má': 4, 'pad': 0, 'space': 1}
```
Comment (horpto) [inline gensim/corpora/dictionary.py]: @menshikh-iv 
not critical at all, but id #5 is lost.

**Pull request merged**